packages/fx-core/src/plugins/solution/fx-solution/arm.ts

```diff
@@ -495,8 +521,19 @@ async function doGenerateArmTemplate(ctx: SolutionContext): Promise<Result<any,
     for (const env of envListResult.value) {
       const parameterFileName = parameterFileNameTemplate.replace(EnvNamePlaceholder, env);
       const parameterEnvFilePath = path.join(parameterEnvFolderPath, parameterFileName);
-      const parameterFileContent = bicepOrchestrationTemplate.getParameterFileContent();
-      await fs.appendFile(parameterEnvFilePath, parameterFileContent);
+      let parameterFileContent = "";
+      if (await fs.pathExists(parameterEnvFilePath)) {
+        const parameterFile = await fs.readJson(parameterEnvFilePath);
+        const paramterObj = parameterFile.parameters.provisionParameters.value;
+        parameterFile.parameters.provisionParameters.value = Object.assign(
+          paramterObj,
+          bicepOrchestrationTemplate.getAppendedParameters()
```

Comment:
not blocking this PR: let's create a task in backlog to figure out the experience the parameter we're going to add already exists in existing azure.parameters.xxx.json

Reply:
sure